front/migrations/20241205_update_space_group_names.ts

```diff
@@ -0,0 +1,44 @@
+import _ from "lodash";
+
+import { Workspace } from "@app/lib/models/workspace";
+import { SpaceResource } from "@app/lib/resources/space_resource";
+import { makeScript } from "@app/scripts/helpers";
+import { Authenticator } from "@app/lib/auth";
+
+async function backfillWorkspacesGroup(execute: boolean) {
+  const workspaces = await Workspace.findAll();
+
+  const chunks = _.chunk(workspaces, 16);
+  for (const [i, c] of chunks.entries()) {
+    console.log(
+      `[execute=${execute}] Processing chunk of ${c.length} workspaces... (${
+        i + 1
+      }/${chunks.length})`
+    );
+    await Promise.all(
+      c.map(async (w) => {
+        const auth = await Authenticator.internalAdminForWorkspace(w.sId);
+        const allSpaces = await SpaceResource.listWorkspaceSpaces(auth);
+        const regularSpaces = allSpaces.filter((s) => s.kind === "regular");
+        for (const space of regularSpaces) {
+          if (space.groups.length === 1) {
```

Comment:
We should probably backfill open space since they still have a group associated. You can loop on the groups associated to the regular space and update the name only if the group is regular, WDYT?

Reply:
:+1: